src/error-once.md

```diff
@@ -0,0 +1,112 @@
+# Handle Errors Once
+
+When a function receives an error from a caller,
+it can handle it in a variety of different ways
+depending on what it knows about the error.
+
+These include, but not are limited to:
+
+- if the function contract defines specific errors,
+  matching the error with `errors.Is` or `errors.As`
+  and handling the branches differently
+- if the error is recoverable,
+  logging the error and degrading gracefully
+- if the error represents a domain-specific failure condition,
+  returning a well-defined error
+- [wrapping the error](error-wrap.md) and returning it
+- returning the error as-is
+
+Regardless of how the error is handled, generally,
+a function should handle an error only once.
+It should not, for example, log the error and then return it again
+because its callers will likely handle the error too.
```

Comment:
oh, by handle you mean "take an action", where action 1 is log, and action 2 is return.

or you mean "a function should touch a received error only once."

Reply:
I meant "take an action". Let me reword a tad bit.